src/js_tests/wirecloud/SourceEndpointSpec.js

```diff
@@ -205,6 +205,71 @@
                 expect(endpoint + "").toEqual([operatorModel.meta.type, operatorModel.id, endpointDesc.name].join("/"));
             });
         });
+
+        describe("toJSON()", function () {
+            var operatorModel, widgetModel;
+
+            beforeAll(function () {
+
+                operatorModel = {
+                    id: "1",
+                    meta: {
+                        type: 'operator'
+                    }
+                };
+
+                widgetModel = {
+                    id: "1",
+                    meta: {
+                        type: 'widget'
+                    }
+                };
+            });
+
+            it("should convert a missing source endpoint to string", function () {
+                var endpointName = "source";
+                var endpointJSON = {
+                    id: widgetModel.id,
+                    type: widgetModel.meta.type,
+                    endpoint: endpointName
+                };
+                var endpoint = new ns.GhostSourceEndpoint(widgetModel, endpointName);
+
+                expect(JSON.parse(JSON.stringify(endpoint))).toEqual(endpointJSON);
+            });
+
+            it("should convert a widget's source endpoint to string", function () {
+                var endpointDesc = {
+                    name: "source",
+                    label: "title",
+                    friendcode: "a b c"
+                };
+                var endpointJSON = {
+                    id: widgetModel.id,
+                    type: widgetModel.meta.type,
+                    endpoint: endpointDesc.name
+                };
+                var endpoint = new ns.WidgetSourceEndpoint(widgetModel, endpointDesc);
+
+                expect(JSON.parse(JSON.stringify(endpoint))).toEqual(endpointJSON);
```

Comment:
This is more a integration test than a unit test. You should check the result of the `toJSON` method.